File: NextApp/iOS/OrgPortalNext/Core/DataLayer/ScheduleRepository.swift
import Foundation
import Model

@MainActor
public protocol ScheduleRepository: AnyObject {
    func fetchAll() async throws -> [Schedule]
    func fetch(on date: Date, calendar: Calendar) async throws -> [Schedule]
    func save(_ schedule: Schedule) async throws
    func delete(id: UUID) async throws
}

File: NextApp/Android/core/data/src/main/kotlin/jp/komehyappyo/member/next/core/data/ScheduleRepository.kt
package jp.komehyappyo.member.next.core.data

import jp.komehyappyo.member.next.core.model.Schedule
import kotlinx.coroutines.flow.Flow
import java.time.LocalDate
import java.util.UUID

interface ScheduleRepository {
    fun observeAll(): Flow<List<Schedule>>
    fun observe(date: LocalDate): Flow<List<Schedule>>
    suspend fun save(schedule: Schedule)
    suspend fun delete(id: UUID)
}
